### PR TITLE
ARM: DNS Zone, Record Set changes & fixes

### PR DESCRIPTION
**Pull request by romangromov**
This PR contains `DNS Zone`, `DNS Record Set` changes in `ARM` requested by @jtuliani :

* using `network dns` and `network dns record-set` style instead of `dns-record-set`, related issue fixed https://github.com/Azure/azure-xplat-cli/issues/2030

* added output for `SOA` Record Set in `network dns record-set list` command, related issue fixed https://github.com/Azure/azure-xplat-cli/issues/2027

* added support for `-z` option in all `record-set` commands, related issue fixed https://github.com/Azure/azure-xplat-cli/issues/2019

* using `azure-arm-dns` package instead of local SDK parts, ~~dnsManagementClient.js~~ removed

* Test cases updated as well to follow new commands style.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@romangromov__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: lib/commands/arm/network/dnsRecordSet._js [hunk 1]
```diff
@@ -126,8 +126,8 @@ __.extend(DnsRecordSet.prototype, {
         self.output.table(outputData, function (row, recordSet) {
           row.cell($('Name'), recordSet.name);
           row.cell($('TTL'), recordSet.properties.ttl);
-          row.cell($('Type'), self._detectType(recordSet.type));
-          row.cell($('Tags'), tagUtils.getTagsInfo(recordSet.tags));
+          row.cell($('Type'), self._detectType(recordSet.id));
```
Comment (yugangw-msft) [inline lib/commands/arm/network/dnsRecordSet._js]: i assume you mean to use ".id", rather ".type", right?

File: test/commands/arm/network/arm.network.dns-zone-record-set-tests.js [hunk 2]
```diff
@@ -61,13 +61,12 @@ describe('arm', function() {
...
-    describe('dns-record-set', function() {
+    describe('dns record-set', function() {
 
-		it('create should create a dns-record-set', function(done) {
-			this.timeout(networkUtil.timeout);
+		it('create should create a dns record-set', function(done) {
 			networkUtil.createGroup(groupName, location, suite, function() {
 				networkUtil.createDnszone(groupName, dnszonePrefix, suite, function() {
-					var cmd = util.format('network dns-record-set create %s %s %s -y %s --json', groupName, dnszonePrefix, dnszoneRecPrefix, Dnstype).split(' ');
+					var cmd = util.format('network dns record-set create %s %s %s -y %s --json', groupName, dnszonePrefix, dnszoneRecPrefix, Dnstype).split(' ');
```
Comment (yugangw-msft) [inline test/commands/arm/network/arm.network.dns-zone-record-set-tests.js]: again, indent

File: lib/commands/arm/network/dnsRecordSet._js (hunk 1 above)
Comment (romangromov) [inline lib/commands/arm/network/dnsRecordSet._js]: @yugangw-msft no, it's exactly `Type` here, we need to retrieve Record Set type from recource `id`

File: test/commands/arm/network/arm.network.dns-zone-record-set-tests.js (hunk 2 above)
Comment (romangromov) [inline test/commands/arm/network/arm.network.dns-zone-record-set-tests.js]: @yugangw-msft thank you, it was initially wrong, now it's fixed

**Pull request merged**